internal/handlers/pg/msg_explain.go

```diff
@@ -67,6 +67,13 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		qp.Sort = nil
 	}
 
+	// Sort requires fetching all documents and sorting them in memory unless sort pushdown is enabled.
+	// Skip with limit requires skip + limit documents which may be more than maximum int64.
```

Comment:
It's not clear from the code how `skip+limit` int64 overflow is handled here for me. :'(

Perhaps the comment should be rephrased?
Or maybe this part of the comment should be located somewhere else, closer to the place where we handle it?

Other than that, this PR looks good to me.

Reply:
Thanks for this, yes this was unclear :see_no_evil:. I rephrased it because skip pushdown is not supported, we can't pushdown limit.